From 47559d11e27beaecec41c1059c215ea61ece62d6 Mon Sep 17 00:00:00 2001
From: Guan Jing <guanjing6@huawei.com>
Date: Wed, 30 Nov 2022 18:23:26 +0800
Subject: [PATCH] sched/fair:ARM64 enables SIS_UTIL and disables SIS_PROP
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 47559d11e27beaecec41c1059c215ea61ece62d6
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I61E4M
CVE: NA

--------------------------------

When doing wakeups, attempt to limit superfluous scans of the LLC domain.
ARM64 enables SIS_UTIL and disables SIS_PROP to search idle CPU based on
sum of util_avg.

Signed-off-by: Guan Jing <guanjing6@huawei.com>
Reviewed-by: Zhang Qiao <zhangqiao22@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/features.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index 7783da2a021d..fef48f5be2fa 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -54,8 +54,13 @@ SCHED_FEAT(TTWU_QUEUE, true)
 /*
  * When doing wakeups, attempt to limit superfluous scans of the LLC domain.
  */
+#ifdef CONFIG_ARM64
 SCHED_FEAT(SIS_PROP, false)
 SCHED_FEAT(SIS_UTIL, true)
+#else
+SCHED_FEAT(SIS_PROP, true)
+SCHED_FEAT(SIS_UTIL, false)
+#endif
 
 #ifdef CONFIG_SCHED_STEAL
 /*
-- 
2.33.0

